src/main/java/com/frcteam3255/components/motors/SN_SparkMax.java

```diff
@@ -0,0 +1,194 @@
+package com.frcteam3255.components.motors;
+
+import com.ctre.phoenix.ErrorCode;
+import com.ctre.phoenix.motorcontrol.DemandType;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.BaseTalonConfiguration;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+
+public class SN_SparkMax extends CANSparkMax implements SN_MotorInterface{
+
+    private SparkMaxPIDController pidController;
+    public RelativeEncoder encoder;
+    private ControlMode pidType = ControlMode.Disabled;
+    private double pidSetpoint;
+
+    /**
+     * Creates a SparkMax object for use with a brushless motor.
+     */
+    public SN_SparkMax(int deviceId) {
+        this(deviceId, MotorType.kBrushless);
+    }
+
+    /**
+     * Creates a SparkMax object with the defined motorType
+     * @param motorType
+	 *            Motor type to use. (Brushed or Brushless)
+     */
+    public SN_SparkMax(int deviceId, MotorType motorType) {
+        super(deviceId, motorType);
+        pidController = super.getPIDController();
+        encoder = super.getEncoder();
+        super.restoreFactoryDefaults();
+        super.burnFlash();
```

Comment:
I don't think we should run `burnFlash()` this often. According to [this](https://github.com/REVrobotics/SPARK-MAX-Server/blob/master/cmd/burnFlash.go#L37) it can cause wear and may only last 10,000 cycles. According to [this](https://docs.revrobotics.com/sparkmax/gs-sm/connect-a-spark-max-over-usb#save-the-settings) it also seems like it might take a few seconds.

The only time this should be useful (I think) is if the Spark Max reboots without the code rebooting, which should never happen because the [roboRIO's minimum input is 7v](https://www.ni.com/docs/en-US/bundle/roborio-frc-specs/page/specs.html) and the [Spark Max's minimum input is 5.5v](https://docs.revrobotics.com/sparkmax/specifications).

Maybe doing this once with desired settings before each comp could be wise, but I really don't think that we need to burn flash at all when controlling over CAN.